### PR TITLE
Additional dbt solids

### PR DESCRIPTION
**Pull request by dehume-drizly**
Was really excited about https://dagster.io/blog/dagster-dbt and seeing the improved Dagster dbt solids. Porting over our Dagster DAG to use these solids, needed the commands to `docs generate` and also `seed`. Only included for the CLI version of dbt because that is what we use. Can also include the these solids for rpc as well for consistency.

### CODE REVIEW

**Comment by chenbobby** [comment]
Thank you, @dehume-drizly for contributing the CLI solids.

I've added solids for `dbt seed` and `dbt docs generate` for RPC as well

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py [hunk 1]
```diff
@@ -535,3 +535,188 @@ def dbt_cli_compile(context) -> Dict:
...
+        "state": Field(
+            config=Noneable([str]),
+            default_value=None,
+            is_required=False,
+            description=(
+                "If set, use the given directory as the source for json files to compare with "
+                "this project."
+            ),
+        ),
+        "yield_materializations": Field(
```
Comment (mgasner) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py:685]: description should be `"If True, materializations corresponding to the results of the dbt operation will be yielded when the solid executes. Default: True"`

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py [hunk 2]
```diff
@@ -535,3 +535,188 @@ def dbt_cli_compile(context) -> Dict:
...
+        "no-version-check": Field(
+            config=bool,
+            description=(
+                "Skip the check that dbt's version matches the one specified in the "
+                "dbt_project.yml file ('require-dbt-version')"
+            ),
+            is_required=False,
+            default_value=False,
+        ),
+        "models": Field(
```
Comment (mgasner) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py:658]: are these config fields relevant to seed?

Comment (chenbobby) [reply python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py:658]: Running `dbt seed -h` shows me that the `--models` flag is actually named `--select`

```
-s MODELS [MODELS ...], --select MODELS [MODELS ...]
                        Specify the nodes to include.
```

it functions similarly to the `--models` flag, but perhaps it should be renamed to `select` for consistency with the official dbt CLI

Comment (dwallace0723) [reply python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py]: @mgasner @chenbobby It looks like there are two config options missing currently form the RPC versions of these solids: `select` and `full_refresh`. We actually already have the `seed` and `docs generate` solids in our project and will need them once we migrate over to using the `dagster-dbt` library, so I'm following this PR just to keep a heads up.

**Comment by mgasner** [comment]
These tests aren't adequate for reasons we discussed @chenbobby -- I can write appropriate tests or you can, as you like.

**Comment by mgasner** [comment]
If we use mocks/snapshots, we need a script to regenerate them from actual dbt runs, or we need to run dbt on a test project as part of the tests. (Prefer the latter as less opportunity for drift). We need to make assertions about the actual semantics of these solids, and we need to exercise the config surface.

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py (hunk 2 above)
Comment (chenbobby) [reply python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py]: Thanks for pointing that out

And, as we discussed @dwallace0723 , the dedicated RPC endpoint for `dbt seed` [doesn't appear to support](https://docs.getdbt.com/reference/commands/rpc#run-seeds) these additional config options, but this can be circumvented by using [the "CLI over RPC" endpoint](https://docs.getdbt.com/reference/commands/rpc#running-a-task-with-cli-syntax)

this would improve more than just this dbt RPC solid, i think. we can create another PR for such a change

**Comment by Torvaney** [comment]
Hi folks,

What is required to get this PR across the line? I would love to use these changes, so happy to put in some time to help out.

Thanks

**Comment by chenbobby** [comment]
Hi @Torvaney, thanks for asking and also for offering to help out 🤗 

From the last time that @mgasner and I discussed (and Max, feel free to correct me if I'm wrong), this PR was put on pause until the testing in `dagster-dbt` was improved. To be more specific, the current test suite has low coverage over dbt's behavior and the wide surface area of configuration that dbt provides. Unfortunately, I think our plates have been quite full and we haven't been able to push improvements lately 😞 

From @mgasner in [a comment above](https://github.com/dagster-io/dagster/pull/3188#issuecomment-724177836):

> We need to make assertions about the actual semantics of these solids, and we need to exercise the config surface.

In my mind a first step would be to add tests that cover additional dbt config. As of right now, most tests use minimal dbt config, like `project-dir` and `profiles-dir`. [Along the lines of testing extra config](https://github.com/dagster-io/dagster/blob/a07fc28ae6878235aeada88e3584c6dea01f845b/python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_solids.py#L39), it would be good to set expectations about how dbt behaves (or fails) based on different permutations of config.

In general, it is inevitable that some updates to dbt will break `dagster-dbt`, but we should at least detect when this happens with tests. At this time, I don't think this is the case.

**Comment by Torvaney** [comment]
Hey @chenbobby 👋. Thanks for your quick reply. 

Just so I'm sure I understand: at the moment, improving the testing of `dagster-dbt` is a higher priority than matching the features of the dbt cli. Is that correct?

Would the best way for me to help here be contributing to the testing, then? Or is it more a matter of being patient and waiting until you folks have a chance to get around to it?

**Comment by chenbobby** [comment]
> Just so I'm sure I understand: at the moment, improving the testing of dagster-dbt is a higher priority than matching the features of the dbt cli. Is that correct?

That's correct, @Torvaney. You're definitely invited to contribute to testing 😄 And feel free to mention/assign me as a reviewer on issues or PRs that you want to open

**Comment by mgasner** [comment]
I think we should defer improvements to testing and just land this tbh
